saleor/product/models.py

```diff
@@ -550,11 +561,13 @@ def annotate_quantities(self):
             ),
         )
 
-    def available_in_channel(self, channel_slug):
-        return self.filter(
-            channel_listings__price_amount__isnull=False,
-            channel_listings__channel__slug=str(channel_slug),
-        )
+    def available_in_channel(self, channel: Optional[Channel]):
+        if not channel:
+            return self.none()
+        channel_listings = ProductVariantChannelListing.objects.filter(
+            price_amount__isnull=False, channel_id=channel.id
+        ).values("id")
```

Comment:
Just a reminder, that in 3.20 port this should also use the `using` method to handle replicas. Not sure if we want this in 3.14 as well.